src/Microsoft.Health.Fhir.Core/Messages/Patch/ConditionalPatchResourceRequest.cs

```diff
@@ -18,18 +18,18 @@ public sealed class ConditionalPatchResourceRequest : ConditionalResourceRequest
 
         public ConditionalPatchResourceRequest(
             string resourceType,
-            JsonPatchDocument patchDocument,
+            PatchPayload payload,
             IReadOnlyList<Tuple<string, string>> conditionalParameters,
             WeakETag weakETag = null)
             : base(resourceType, conditionalParameters)
         {
-            EnsureArg.IsNotNull(patchDocument, nameof(patchDocument));
+            EnsureArg.IsNotNull(payload, nameof(payload));
 
-            PatchDocument = patchDocument;
+            Payload = payload;
             WeakETag = weakETag;
         }
 
-        public JsonPatchDocument PatchDocument { get; }
+        public PatchPayload Payload { get; }
```

Comment:
I think this is good, always felt a little strange having the asp.net JsonPatchDocument here directly :)

Reply:
Agreed! Big thanks to Ivan for the help and contributions to the PR by pushing us in the right direction.